File: SDKs/microsoft.com/cognitiveservices-Training/1.2/go/pkg/models/operations/createproject.go
package operations

import (
	"openapi/pkg/models/shared"
)

type CreateProjectQueryParams struct {
	Description *string `queryParam:"style=form,explode=true,name=description"`
	DomainID    *string `queryParam:"style=form,explode=true,name=domainId"`
	Name        string  `queryParam:"style=form,explode=true,name=name"`
}

type CreateProjectHeaders struct {
	TrainingKey string `header:"style=simple,explode=false,name=Training-Key"`
}

type CreateProjectRequest struct {
	QueryParams CreateProjectQueryParams
	Headers     CreateProjectHeaders
}

type CreateProjectResponse struct {
	Body        []byte
	ContentType string
	Project     *shared.Project
	StatusCode  int64
}

File: SDKs/microsoft.com/cognitiveservices-Training/2.0/typescript/src/sdk/models/operations/createproject.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class CreateProjectQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=description" })
  description?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=domainId" })
  domainId?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=name" })
  name: string;
}


export class CreateProjectHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=Training-Key" })
  trainingKey: string;
}


export class CreateProjectRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: CreateProjectQueryParams;

  @Metadata()
  headers: CreateProjectHeaders;
}


export class CreateProjectResponse extends SpeakeasyBase {
  @Metadata()
  body?: Uint8Array;

  @Metadata()
  contentType: string;

  @Metadata()
  project?: shared.Project;

  @Metadata()
  statusCode: number;
}
